packages/fuels-abigen-macro/src/lib.rs

```diff
@@ -21,7 +26,13 @@ pub fn abigen(input: TokenStream) -> TokenStream {
 pub fn wasm_abigen(input: TokenStream) -> TokenStream {
     let args = parse_macro_input!(input as Spanned<ContractArgs>);
 
-    let c = Abigen::new(&args.name, &args.abi).unwrap().no_std();
+    let c = Abigen::new(
+        &args.name,
+        &args.abi,
+        args.binary_filepath.as_ref().map(|v| v.as_ref()),
```

Comment:
not a big fan of this `.map()` call, it seems weird to me

Reply:
@vnepveu what's wrong with the map function? do you have any cleaner way to _map_ `Option<T>` to `Option<U>` 🤔

Reply:
I  think I'm just not used to seeing `map` being used on something that is not iterable and only has one element. But if that's the only way then let's go!

Reply:
it's a different type of `map` the one that works on `Option<T>` types:

https://doc.rust-lang.org/std/option/enum.Option.html#method.map